x-pack/platform/plugins/shared/maps/public/routes/map_page/map_app/map_app.tsx

```diff
@@ -578,10 +580,28 @@ export class MapApp extends React.Component<Props, State> {
     }
 
     return (
-      <div id="maps-plugin" className={this.props.isFullScreen ? 'mapFullScreen' : ''}>
+      <div
+        id="maps-plugin"
+        css={css`
```

Comment:
For performance, avoid composing styles directly inside the css\`` template. Instead, define static and conditional styles outside, and pass them as an array. This prevents re-serializing the template on each render and improves readability.

```
const mapsPluginStyles = {
  display: 'flex',
  flexDirection: 'column',
  width: '100%',
  overflow: 'hidden',
};

const fullscreenStyles = { height: '100vh !important' };

css={[kibanaFullBodyHeightCss(), mapsPluginStyles, isFullScreen && fullscreenStyles]}
```

(reference read: https://douges.dev/blog/taming-the-beast-that-is-css-in-js)